File: java/duck/ducks/RubberDuck.java
package ducks;

import behaviors.fly.FlyWithWings;
import behaviors.quack.Squeak;

public class RubberDuck extends Duck {

    public RubberDuck(String name) {
        super(name);
        flyBehavior = new FlyWithWings(name);
        quackBehavior = new Squeak(name);
    }
}

File: javascript/duck/ducks/RubberDuck.js
import {generalBehavior} from "../behaviors/GeneralBehavior.js";
import {flyBehavior} from "../behaviors/FlyBehavior.js";
import {quackBehavior} from "../behaviors/QuackBehavior.js";

export function RubberDuck(name) {
    const duck = { name: name }
    return {
        ...duck,
        ...generalBehavior(duck),
        ...flyBehavior(duck),
        ...quackBehavior(duck)
    }
}